.github/workflows/terraform.yml

```diff
@@ -111,7 +115,7 @@ jobs:
 
       - name: Terraform plan
         uses: dflook/terraform-plan@0f4a5046f18698eee737a52d4cb11f56f2377a32 # v1.42.0
-        if: github.event_name == 'pull_request'
+        if: github.event_name == 'pull_request' || inputs.terraform_operation == 'apply'
```

Comment:
I think it's not needed, as per docs it would just print plan on stdout outside of a pull_request event, but then the apply would be immediately executed after - without using that

https://github.com/dflook/terraform-plan?tab=readme-ov-file#terraform-plan-action

Reply:
https://github.com/dflook/terraform-github-actions/tree/main/terraform-apply 
Without a plan apply will fail. Have to add auto_approve option based on event name.

Reply:
it could still work if the latest commit is a PR with a plan attached to it - but in that case you could simply re-run the merge workflow run - no need to trigger this new workflow.

anyway could be good to have it in case we know we want to run an apply.